nats/src/main.rs

```diff
@@ -23,18 +23,25 @@ const ENV_NATS_CLIENT_SEED: &str = "CLIENT_SEED";
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     // handle lattice control messages and forward rpc to the provider dispatch
     // returns when provider receives a shutdown control message
-    provider_main(
-        NatsMessagingProvider::default(),
-        Some("Nats Messaging Provider".to_string()),
-    )?;
+    let host_data = load_host_data()?;
+    let provider = if let Some(c) = host_data.config_json.as_ref() {
+        let config: ConnectionConfig = serde_json::from_str(c)?;
```

Comment:
I missed this before, but since an error here will halt the provider, let's add an `eprintln!` indicating what went wrong (note we can't log here, since `provider_main` hasn't been called yet 😔

Reply:
since we're using `?` here, what about an `expect`, which will dump to stderr during said panic?

Reply:
Works for me!